akka-actor-typed/src/main/scala/akka/actor/typed/Props.scala

```diff
@@ -210,3 +210,60 @@ private[akka] final case class DispatcherFromConfig(path: String, next: Props =
     extends DispatcherSelector {
   override def withNext(next: Props): Props = copy(next = next)
 }
+
+/**
+ * Not for user extension.
+ */
+@DoNotInherit
+sealed abstract class MailboxSelector extends Props
+
+object MailboxSelector {
+
+  private val _default = DefaultMailboxSelector()
+
+  /**
+   * The default mailbox is unbounded and backed by a [[java.util.concurrent.ConcurrentLinkedQueue]]
+   */
+  def defaultMailbox(): MailboxSelector = _default
+
+  /**
+   * A mailbox with a max capacity after which new messages are dropped (passed to deadletters)
+   * @param capacity The maximum number of messages in the mailbox before new messages are dropped
+   */
+  def bounded(capacity: Int): MailboxSelector = BoundedMailboxSelector(capacity)
+
+  /**
+   * Select a mailbox from the config file using an absolute config path.
+   *
+   * This is a power user settings default or bounded should be preferred unless you know what you are doing.
+   */
+  def fromConfig(path: String): MailboxSelector = MailboxFromConfigSelector(path)
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] final case class DefaultMailboxSelector(next: Props = Props.empty) extends MailboxSelector {
```

Comment:
Curious what the use case is for the empty Props, where default args are needed.

Reply:
`Props.empty.withDispatcher(...).withMailbox(..)`